sound/soc/sof/sof-priv.h

```diff
@@ -193,7 +193,8 @@ struct snd_sof_dsp_ops {
 	int (*runtime_idle)(struct snd_sof_dev *sof_dev); /* optional */
 	int (*set_hw_params_upon_resume)(struct snd_sof_dev *sdev); /* optional */
 	int (*set_power_state)(struct snd_sof_dev *sdev,
-			       const struct sof_dsp_power_state *target_state); /* optional */
+			       const struct sof_dsp_power_state *target_state,
+			       bool trace_enable); /* optional */
```

Comment:
@ranj063 The "trace_enable" is a bit out of place. It seems it's only used to relay info whether system is in S0 or not. So why not just pass "system_target_state" as an additional param here and have the trace specific logic completely in hda-dsp.c implementation. Now the logic that "S0 implies traces can be enabled" is encoded in multiple places.

Reply:
@kv2019i I am not sure I understood your point completely. When entering D0i3, there are 2 options: keep trace enabled or disable it (which is done in the FW). Also, this is not tied with the system_suspend_target because you could enter D0i3 when you are not suspending the system too.

Reply:
or are you just saying that we should use SYSTEM_SUSPEND_NONE as the criterion to determine that trace should be kept enabled?

Reply:
fixed now